openmdao/jacobians/dictionary_jacobian.py

```diff
@@ -166,6 +167,12 @@ class _CheckingJacobian(DictionaryJacobian):
     def __init__(self, system):
         super().__init__(system)
         self._subjacs_info = self._subjacs_info.copy()
+
+        # Convert any scipy.sparse subjacs to dense.
+        for key, subjac in self._subjacs_info.items():
+            if sp.issparse(subjac['val']):
+                self._subjacs_info[key]['val'] = subjac['val'].toarray()
```

Comment:
This will definitely work, but in most cases if a user is using sparse matrices it's because the dense version would be really big, so I'm not sure we want to do this. What if we just converted subjac['val'] to coo format and then set 'rows' and 'cols' metadata based on the coomat.row and coomat.col and set the subjac['val'] based on coomat.data?  That would use a lot less memory and we'd still be checking that the sparsity was correct.

Reply:
yeah, that's probably wiser. I've made the change.

Reply:
Yeah, that's probably wiser.  Made the change.